src/main/java/by/intexsoft/customers/products/MapToList.java

```diff
@@ -0,0 +1,34 @@
+package by.intexsoft.customers.products;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class MapToList {
+    public static List<List<String>> transformMapToList(
+            Map<String, Long> customersProducts, long threshold) {
+        if (customersProducts == null) {
+            throw new IllegalArgumentException("Map can't be null!");
+        }
+        List<List<String>> listOfCustomers = new ArrayList<>();
+        final List<String> tempCustomersList = new ArrayList<>();
+        long currentSum = 0;
+        for (final Map.Entry<String, Long> entry : customersProducts.entrySet()) {
+            tempCustomersList.add(entry.getKey());
+            currentSum += entry.getValue();
+            if (currentSum > threshold && tempCustomersList.size() > 1) {
+                listOfCustomers.add(new ArrayList<String>() {{
+                    addAll(0, tempCustomersList.subList(0, tempCustomersList.size() -1));}});
```

Comment:
All is ok, but I think that we can simplify it.
Current realization always adds current element to tempCustomersList and increases currentSum (lines 17, 18).
So here(line 21) we have to "remove" last element. Lets try to refactor this code and add elements in tempCustomersList only if it is realy necessary.

Reply:
Ok, it seems like a bad idea at a glance, but somehow for me only this one worked. I'll try to change the logic